src/python_routing_v01/compute_nhd_routing_SingleSeg.py

```diff
@@ -302,52 +322,256 @@ def compute_mc_reach_up2down(
+# arguments network, number of timesteps (nts),  timestep in seconds  (dt),  pathToOutputFile
+# using global connections and flowveldepth.
+def writeArraytoNC(
+    network=None,
+    nts=0,
+    dt=60,
+    verbose=False,
+    debuglevel=0,
+    pathToOutputFile="../../test/output/text/",
+):
+    global connections
+    global flowveldepth
+    # create  array variables to copy from python "flowveldepth" which is global
+    flowveldepth_data = {
+        "segment": [],
+        "time": [],
+        "qlatval": [],
+        "flowval": [],
+        "depthval": [],
+        "velval": [],
+    }
+
+    ordered_reaches = {}
+    for head_segment, reach in network["reaches"].items():
+        if reach["seqorder"] not in ordered_reaches:
+            ordered_reaches.update(
+                {reach["seqorder"]: []}
+            )  # TODO: Should this be a set/dictionary?
+        ordered_reaches[reach["seqorder"]].append([head_segment, reach])
+
+    # get data into array - preparation step
+    TIME_WRITTEN = False
+    write_segment = None
+    for x in range(network["maximum_reach_seqorder"], -1, -1):
+        for head_segment, reach in ordered_reaches[x]:
+            current_segment = reach["reach_head"]
+            while True:
+                # appending data from each segments to a single list  "flowveldepth_data"
+                # preserving ordering same as segment in a reach
+                flowveldepth_data["qlatval"].append(
+                    flowveldepth[current_segment]["qlatval"]
+                )
+                flowveldepth_data["flowval"].append(
+                    flowveldepth[current_segment]["flowval"]
+                )
+                flowveldepth_data["depthval"].append(
+                    flowveldepth[current_segment]["depthval"]
+                )
+                flowveldepth_data["velval"].append(
+                    flowveldepth[current_segment]["velval"]
+                )
+                # write segment flowveldepth_data['segment']
+                flowveldepth_data["segment"].append(current_segment)
+                if not TIME_WRITTEN:
+                    # write time only once - for first segment
+                    flowveldepth_data["time"].append(
+                        flowveldepth[current_segment]["time"]
+                    )
+                    TIME_WRITTEN = True
+
+                if current_segment == reach["reach_tail"]:
+                    write_segment = current_segment
+                    if debuglevel <= -2:
+                        print(f"{current_segment} (tail)")
+                    break
+                next_segment = connections[current_segment]["downstream"]
+                if debuglevel <= -3:
+                    print(f"{current_segment} --> {next_segment}\n")
+                current_segment = next_segment
+
+    # check number of timesteps should match the time the data is written
+    if int(len(flowveldepth_data["time"][0])) != int(nts):
```

Comment:
`len` always returns an int. We don't need to cast here.

Reply:
@jdhondia-noaa was being very careful. 

nts (number of time steps) should always be an integer, too, though. 

Really, if they are not equal, we definitely want an error, even if the error arises from strange typing, because it would be just as much of a problem, so I removed the casting.